### ISSUE TITLE
How to disable COC for diff?

### ISSUE THREAD

**Issue by ronakg** [neoclide/coc.nvim#1025]
I've tried following in my `vimrc` but it's not recognizing `CocDisable` as a command for some reason.

```
if &diff
    CocDisable
endif
```

**Comment by chemzqm** [comment]
Use autocmd with `:h b:coc_enabled`

**Comment by ronakg** [comment]
Thanks.

**Comment by ronakg** [comment]
I never got back to this. The problem is that during `BufCreate` the `filetype` is not set yet. So I can't check during `BufCreate` if the `filetype` is set to `diff` or not.

`let b:coc_enabled = 0` only works if it's during `BufCreate`.

Do you have any other workaround?

**Comment by chemzqm** [comment]
Use `autocmd FileType html` instead.

**Comment by ronakg** [comment]
Can you please elaborate?

I tried the following, both didn't work. I want to disable CoC completely when I open diff.

```vim
        autocmd FileType html CocDisable
```
and 

```vim
        autocmd FileType html let b:coc_enabled = 0
```

**Comment by oblitum** [comment]
If you mean vim diff mode, check whether `&diff` is enabled, for example put `let b:coc_enabled = &diff` on your `.vimrc`.

**Comment by ronakg** [comment]
> If you mean vim diff mode, check whether `&diff` is enabled, for example put `let b:coc_enabled = !&diff` on your `.vimrc`.

This doesn't work either. Problem is that `b:coc_enabled` is useful only if it's set to `0` before the buffer is created. `&diff` is set after the buffer is created.

**Comment by ronakg** [comment]
Any ideas?

**Comment by oblitum** [comment]
> So I can't check during BufCreate if the filetype is set to diff or not

`&diff` is not a filetype, see `:h diff`:

```
In your vimrc file you could do something special when Vim was started in
diff mode.  You could use a construct like this: >

	if &diff
	   setup for diff mode
	else
	   setup for non-diff mode
	endif
```

**Comment by oblitum** [comment]
If it's a must for you, you may also go the route of lazy load coc (have it globally disabled, and enable it on later events).

**Comment by ronakg** [comment]
> If it's a must for you, you may also go the route of lazy load coc (have it globally disabled, and enable it on later events).

This did the trick. I had to move everything related to CoC inside `if !&diff` including loading the plugin itself to keymaps, functions, statusline integrations, etc. Not ideal, but it works.

**Comment by yusuphgammut** [comment]
> Use autocmd with `:h b:coc_enabled`

Let me add to the conversation the solution that worked for me, for future visitors of this thread:

```
augroup disableCocInDiff
  autocmd!
  autocmd DiffUpdated * let b:coc_enabled=0
augroup END
```

I'm using [fugitive.vim](https://github.com/tpope/vim-fugitive) to enhance my diffs but when I added coc.nvim to my neovim configuration I started noticing that eslint was failing when triggering a diff on a file. This code snippet creates an autocommand group that wraps an autocommand that is triggered after a diff is updated and creates the variable suggested by @chemzqm . Works perfectly for me!

Hope it helps.
